### PR TITLE
[Enhancement] Infer JAVA_HOME if not set

### PR DESCRIPTION
**Pull request by sduzh**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: env.sh [hunk 1]
```diff
@@ -68,7 +68,9 @@ fi
 export CLANG_COMPATIBLE_FLAGS=`echo | ${STARROCKS_GCC_HOME}/bin/gcc -Wp,-v -xc++ - -fsyntax-only 2>&1 \
                 | grep -E '^\s+/' | awk '{print "-I" $1}' | tr '\n' ' '`
 
-# check java home
+if [[ -z ${JAVA_HOME} ]]; then
+    JAVA_HOME="$(dirname $(dirname $(readlink -f $(which javac))))"
```
Comment (imay) [inline env.sh:72]: Better to print a log to indicate that this JAVA_HOME is inferred.

Comment (kevincai) [reply env.sh:72]: need export JAVA_HOME

Comment (sduzh) [reply env.sh:72]: updated

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request closed without merging**